ui/dashboards/src/context/DashboardAppSlice.tsx

```diff
@@ -0,0 +1,45 @@
+import { immer } from 'zustand/middleware/immer';
+import { useDashboardStore } from './DashboardProvider';
+
+export interface DashboardAppSlice {
+  addPanelComponent: {
+    isOpen: boolean;
+    groupIndex: number;
+    setIsOpen: (isOpen: boolean, groupIndex?: number, panelRef?: string) => void;
+    panelRef?: string;
+  };
+  addGroupComponent: {
+    isOpen: boolean;
+    setIsOpen: (isOpen: boolean, index?: number) => void;
+    index?: number;
+  };
+}
+
+export const createDashboardAppSlice = immer<DashboardAppSlice>((set) => ({
+  addPanelComponent: {
+    isOpen: false,
+    groupIndex: 0,
+    setIsOpen: (isOpen: boolean, groupIndex = 0, panelRef?: string) =>
+      set((state) => {
+        state.addPanelComponent.isOpen = isOpen;
+        state.addPanelComponent.groupIndex = groupIndex;
+        state.addPanelComponent.panelRef = panelRef;
+      }),
+  },
+  addGroupComponent: {
+    isOpen: false,
+    isEdit: false,
+    setIsOpen: (isOpen: boolean, index?: number) =>
```

Comment:
similar to add panel component, we also need to know when the add group component is used for adding a new panel group vs editing an existing panel group